### PR TITLE
feat: Add Home Page and user authentication

### PR DESCRIPTION
**Pull request by Naman-Saxena1**
# Added Home and Auth Pages
1. Created basic project structure
2. Configured Material UI, Redux store, react-toastify, heroku and MongoDb database
3. User can create account and login to site

## Files to review
1. GenreCard.tsx in components folder
- This component is used to show genres on Home Page

2. Home.tsx in pages folder
- This component is used to represent Home Page of site

Note:
1. You can ignore Auth Files like Signup.tsx and Login.tsx because they are just TypeScript version of files that have already been reviewed in all my previous projects.
2. You can choose to ignore Redux files

## Home Page
Home1
Home2

## Signup Page
Signup

## Login Page
Login

### CODE REVIEW

File: src/components/GenreCard/GenreCard.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import {
+    Button
+} from "@mui/material"
+
+type GenreCardProps = {
+    imgSrc: string
+    genreName: string
+    noOfQuizzesAvailable: number
+}
```
Comment (gopalbbharadva) [inline src/components/GenreCard/GenreCard.tsx]: Suggestion: I made one folder called **types** which contain all type files with suffix types. it's easy to manageable all types.

File: src/pages/AuthenticationPages/Signup.tsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            {
+                newUserName: `${newUserName}`,
+                newUserEmail: `${newUserEmail}`,
+                newUserPassword : `${newUserPassword}`
+            }
+        )
+        .then(res => {
+            if(res.data.status==='ok')
+            {
+                //User created successfully, navigate to Login Page
```
Comment (gopalbbharadva) [inline src/pages/AuthenticationPages/Signup.tsx]: Remove these comments.

**Review by gopalbbharadva — APPROVED** [review]
UI looks nice, Good to go.

**Review by mohammed-sas — APPROVED** [review]
Authentication is working as intended
Overall code is clean

File: src/components/GenreCard/GenreCard.tsx (hunk 1 above)
Comment (Naman-Saxena1) [reply src/components/GenreCard/GenreCard.tsx]: Thanks for suggestion, will make this change before merging 😄

File: src/pages/AuthenticationPages/Login.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      {
+        setNewPasswordType("text")
+      }
+      else
+      {
+        setNewPasswordType("password")
+      }
+    }
+
+    function loginUser(event: React.FormEvent)
```
Comment (VedanthB) [inline src/pages/AuthenticationPages/Login.tsx:39]: Suggestion:  You can move all the api calls into a services folder, and also move the loginUser in to maybe utils/authUtils

**Review by VedanthB — APPROVED** [review]

**Pull request merged**